src/server/main_service.cc

```diff
@@ -1446,24 +1446,26 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
 
 void Service::Publish(CmdArgList args, ConnectionContext* cntx) {
   string_view channel = ArgS(args, 0);
+  string_view msg = ArgS(args, 1);
 
   auto* cs = ServerState::tlocal()->channel_store();
   vector<ChannelStore::Subscriber> subscribers = cs->FetchSubscribers(channel);
   int num_published = subscribers.size();
 
   if (!subscribers.empty()) {
     auto subscribers_ptr = make_shared<decltype(subscribers)>(move(subscribers));
-    auto msg_ptr = make_shared<string>(ArgS(args, 1));
-    auto channel_ptr = make_shared<string>(channel);
+    auto buf = shared_ptr<char[]>{new char[channel.size() + msg.size()]};
```

Comment:
nit: make_shared saves you an allocation.

Reply:
Its c++ 20 👿https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared